### PR TITLE
Next onboarding lesson: NFCorpus

### PR DESCRIPTION
**Pull request by lintool**
@sahel-sh can you try running through this next lesson?

Like before, CR to provide comments, and then a separate PR to add to repro log.

### CODE REVIEW

File: docs/experiments-nfcorpus.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+The final step, as with Lucene, is to learn to use the dense retriever _interactively_.
+This contrasts with the _batch_ run above.
+
+Here's the snippet of Python code that does what we want:
+
+```python
+from pyserini.search.faiss import FaissSearcher, AutoQueryEncoder
+
+encoder = AutoQueryEncoder('facebook/contriever-msmarco', device='cpu', pooling='mean')
+searcher = FaissSearcher('indexes/faiss.nfcorpus.contriever-msmacro.v1', encoder)
```
Comment (sahel-sh) [inline docs/experiments-nfcorpus.md:155]:
```suggestion
searcher = FaissSearcher('indexes/faiss.nfcorpus.contriever-msmacro', encoder)
```

File: docs/experiments-nfcorpus.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+hits = searcher.search('How to Help Prevent Abdominal Aortic Aneurysms')
+
+for i in range(0, 10):
+    print(f'{i+1:2} {hits[i].docid:7} {hits[i].score:.5f}')
```
Comment (sahel-sh) [inline docs/experiments-nfcorpus.md:159]:
```suggestion
    print(f'{i+1:2} {hits[i].docid:7} {hits[i].score:.6f}')
```

File: docs/experiments-nfcorpus.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+```
+
+The `FaissSearcher` provides search capabilities using Faiss as its underlying implementation.
+The `AutoQueryEncoder` allows us to intialize an encoder using a HuggingFace model.
+
+```
+ 1 MED-4555 1.47220
```
Comment (sahel-sh) [inline docs/experiments-nfcorpus.md:166]:
```suggestion
 1 MED-4555 1.472201
```

File: docs/experiments-nfcorpus.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+ 2 MED-3180 1.12501
```
Comment (sahel-sh) [inline docs/experiments-nfcorpus.md:167]:
```suggestion
 2 MED-3180 1.125014
```

File: docs/experiments-nfcorpus.md [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+ 3 MED-1309 1.06715
```
Comment (sahel-sh) [inline docs/experiments-nfcorpus.md:168]:
```suggestion
 3 MED-1309 1.067153
```

File: docs/experiments-nfcorpus.md [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+ 4 MED-2224 1.05954
```
Comment (sahel-sh) [inline docs/experiments-nfcorpus.md:169]:
```suggestion
 4 MED-2224 1.059537
```

File: docs/experiments-nfcorpus.md [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+ 5 MED-4423 1.03844
```
Comment (sahel-sh) [inline docs/experiments-nfcorpus.md:170]:
```suggestion
 5 MED-4423 1.038440
```

File: docs/experiments-nfcorpus.md [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+ 6 MED-4887 1.03262
```
Comment (sahel-sh) [inline docs/experiments-nfcorpus.md:171]:
```suggestion
 6 MED-4887 1.032622
```

File: docs/experiments-nfcorpus.md [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+ 7 MED-2530 1.02076
```
Comment (sahel-sh) [inline docs/experiments-nfcorpus.md:172]:
```suggestion
 7 MED-2530 1.020758
```

File: docs/experiments-nfcorpus.md [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+ 8 MED-2372 1.01614
```
Comment (sahel-sh) [inline docs/experiments-nfcorpus.md:173]:
```suggestion
 8 MED-2372 1.016142
```

File: docs/experiments-nfcorpus.md [hunk 11]
```diff
@@ -0,0 +1,199 @@
...
+ 9 MED-1006 1.01360
```
Comment (sahel-sh) [inline docs/experiments-nfcorpus.md:174]:
```suggestion
 9 MED-1006 1.013599
```

File: docs/experiments-nfcorpus.md [hunk 12]
```diff
@@ -0,0 +1,199 @@
...
+10 MED-2587 1.01081
```
Comment (sahel-sh) [inline docs/experiments-nfcorpus.md:175]:
```suggestion
10 MED-2587 1.010811
```

File: docs/experiments-nfcorpus.md [hunk 13]
```diff
@@ -0,0 +1,197 @@
...
...sed the Cox proportional hazards regression method to estimate mortality among statin users with statin use as time-dependent variable. A total of 4,151 participants had used statins. During the median follow-up of 3.25 years after the diagnosis (range 0.08\u20139.0 years) 6,011 participants died, of which 3,619 (60.2%) was due to breast cancer. After adjustment for age, tumor characteristics, and treatment selection, both post-diagnostic and pre-diagnostic statin use were associated with lowered risk of breast cancer death (HR 0.46, 95% CI 0.38\u20130.55 and HR 0.54, 95% CI 0.44\u20130.67, respectively). The risk decrease by post-diagnostic statin use was likely affected by healthy adherer bias; that is, the greater likelihood of dying cancer patients to discontinue statin use as the association was not clearly dose-dependent and observed already at low-dose/short-term use. The dose- and time-dependence of the survival benefit among pre-diagnostic statin users suggests a possible causal effect that should be evaluated further in a clinical trial testing statins\u2019 effect on survival in breast cancer patients.", "metadata": {"url": "http://www.ncbi.nlm.nih.gov/pubmed/25329299"}}
```
[the remark this answers is not in the document]
Comment (lintool) [reply docs/experiments-nfcorpus.md:51]: Your suggestion is to make the output more readable? I was thinking it's more important to preserve exactly the command output?

File: docs/experiments-nfcorpus.md (hunk 8 above)
Comment (lintool) [reply docs/experiments-nfcorpus.md:171]: Hrm... I think these are minor floating point rounding issues based on different hardware, OS, etc. Probably not going to match _exactly_ for everyone? Maybe add a note to that effect?

File: docs/experiments-nfcorpus.md (hunk 2 above)
Comment (lintool) [reply docs/experiments-nfcorpus.md:159]: Why this suggestion?

File: docs/experiments-nfcorpus.md (hunk 8 above)
Comment (lintool) [reply docs/experiments-nfcorpus.md:171]: Ah, reading your comments out of order. Why more precision?

File: docs/experiments-nfcorpus.md [hunk 14]
```diff
@@ -0,0 +1,197 @@
...
+```
+
+Similarly, we need to munge the relevance judgments (qrels) into the right format.
+This command-line invocation does the trik:
+
+```bash
+tail -n +2 collections/nfcorpus/qrels/test.tsv | sed 's/\t/\tQ0\t/' > collections/nfcorpus/qrels/test.qrels
+```
+
+Okay, the data are ready now.
```
[the remark this answers is not in the document]
Comment (lintool) [reply docs/experiments-nfcorpus.md:74]: I personally write treating data as plural :)

**Comment by lintool** [comment]
@sahel-sh okay, let's try again?

File: docs/experiments-nfcorpus.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (sahel-sh) [reply docs/experiments-nfcorpus.md:51]: sure, either way is fine

File: docs/experiments-nfcorpus.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (sahel-sh) [reply docs/experiments-nfcorpus.md:74]: got it!

**Review by sahel-sh — APPROVED** [review]
LGTM, Thank you!

**Pull request merged**